### ISSUE TITLE
Impossible to train a covolutional autoencoder

### ISSUE THREAD

**Issue by JSharp4273** [skorch-dev/skorch#817]
I want to train the following model:

`model = nn.Sequential(nn.Conv1d(in_channels=2203, out_channels=12, kernel_size=1), nn.BatchNorm1d(12), nn.SELU())`

I am using this autoencoder as dimensionality reduction method.
I am processing a matrix of size 2203 x 233760. 
After the processing I am expecting a matrix of size 12 x 233760.

When I am not using the `NeuralNet` class, I can train my model like this:
`y = model(to_tensor(X.T[:,:,np.newaxis],'cpu'))`

When I am creating an instance of the `NeuralNet` class as follows:
`model_nn = NeuralNet(model, nn.MSELoss, device='cpu')`
I do not arrive to fit my data in order to have the output dimension I do expect.

What do I do wrong ?

**Comment by JSharp4273** [comment]
A part of the problem is in the train_split parameter, however I was able to fix my issue by creating my model as:
`model_nn = NeuralNet(model, nn.MSELoss, device='cpu', train_split=CVSplit(2))`
Then I could successfully fit it:
`XX = X[np.newaxis,:,:]`
`model_nn.fit(np.concatenate([XX] * 2, 0))`
And predict:
`y = model_nn.predict(XX)`
Note the number of element in the concatenation is related to the `train_split` parameters.

**Comment by BenjaminBossan** [comment]
> After the processing I am expecting a matrix of size 12 x 233760.

For the validation split, it looks like you have only 12 samples, which is probably why it failed. But I assume that 233760 is your actual sample size. So another solution that would allow you to keep the normal split is to transpose your data so that the sample dimension is the 0th dimension.

**Comment by JSharp4273** [comment]
Thank you for your post.

No in fact I have 2203 samples, which I want to reduce to 12.
The 233760 is in fact a number of pixel, while the 2203 is the number of images.

For several days I made may trials, with different organisation
Except the solution that I posted before nothing had worked when I provide the model to the class `NeuralNet`.

**Comment by BenjaminBossan** [comment]
Okay, thx for clarifying. So if you have really one 12 samples at the end, you might consider turning off the validation split completely, otherwise only 6 of your 12 samples are used for training and 6 for validation. You can do that by setting `train_split=None`.

**Comment by JSharp4273** [comment]
Thank you for the information.
